validator/api/src/main/java/tech/pegasys/teku/validator/api/GraffitiParser.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.api;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.tuweni.bytes.Bytes32;
+
+public class GraffitiParser {
+  public static Bytes32 loadFromFile(final Path graffitiFile) throws GraffitiLoaderException {
+    try {
+      checkNotNull(graffitiFile, "GraffitiFile path cannot be null");
+      if (Files.size(graffitiFile) > 32) {
+        throw new GraffitiLoaderException("GraffitiFile size too big, maximum size is 32 bytes");
+      }
+      try (final InputStream graffitiFileInputStream = Files.newInputStream(graffitiFile)) {
+        return Bytes32Parser.toBytes32(strip(graffitiFileInputStream.readNBytes(32)));
+      }
+    } catch (final FileNotFoundException e) {
+      throw new GraffitiLoaderException("GraffitiFile file not found: " + graffitiFile, e);
+    } catch (final IOException e) {
+      throw new GraffitiLoaderException(
+          "Unexpected IO error while reading GraffitiFile: " + e.getMessage(), e);
+    }
+  }
+
+  private static byte[] strip(final byte[] value) {
+    return new String(value, StandardCharsets.UTF_8).strip().getBytes(StandardCharsets.UTF_8);
+  }
```

Comment:
this is all just really a question and some thoughts to ponder:
this puts an interesting twist on the concept of 'file is too big' really... If it's a file with a bunch of whitespace that puts it over, this would strip that whitespace and you might be left with valid graffiti? I guess what I'm thinking is the poor person that has '\n' at the end of their file that pushes them over the graffiti limit, but equally accidentally putting whitespace before?
I'm not sure what the answer is, just a question, because really if we're stripping we could potentially not check file size up front but rather attempt to get 32 bytes of graffiti from the file.
so realistic use cases 
 - `bytes32\n`
 - ` bytes32` (leading space)
 - `\nbytes32\n`

Currently these stripped characters would be basically 'your loss', I guess, you have a \n at the end, so only get 31 bytes of graffiti, but this does seem like you'd potentially have a frustrating time figuring out where that byte disappears...

Reply:
Lovely 👍 

I think I'd prefer to do the kindest thing for well meaning users, so I think we should accommodate the "32 bytes and a[n accidental] new line character" scenario.

One concern Meredith raised was that we want to fail fast and if possible avoid reading a totally inappropriate file into memory at all.

In their current form I think these ideas are mutually exclusive - either we read the file to strip the whitespace to measure the length of the bytes that are left or we don't and we rely on the file metadata supplied by the OS which gives us the length inc any whitespace.

A hybrid solution could be to test for a file of 40 bytes or less, and consider it unreasonable for a user to supply more than 8 bytes of leading or trailing whitespace. Once we've read the file, use the existing implementation here which will strip and convert to a Bytes32 (or fail). This allows users to make minor mistakes but also maintains our shared desire to not read files which are obviously too large or the wrong file altogether.

How does that sound?

For example, with a couple of tweaks to some tests, something like this:
```
      if (Files.size(graffitiFile) > 40) {
        throw new GraffitiLoaderException("GraffitiFile size too big, maximum size for graffiti is 32 bytes");
      }
```

Reply:
Yep i think reading a few extra bytes and allowing the fast fail is still good, and it just means we need to check the graffiti is only 32 bytes once we're done and potentially throw the error at the end too...
I completely agree that reading a large buffer wouldn't be great, I was thinking just being nice and allowing a few bytes either side, so this is good.

The other scenario (i think there's a ticket somewhere!) is to read binary from file....
In that scenario we wouldn't be able to do this conversion at all, so it'd become a moot point and we'd just read the 32 bytes max and move on, but i think given this ticket is looking at text, we get that in and worry about the other scenario later...

Reply:
Sounds like a good plan @EdwardPrentice.  I think the main thing we want to avoid is reading in an excessive amount of data, but reading in even 64 bytes and then cleaning up / validating sounds good to me 👍

Reply:
Ok awesome. Will go for 40 and can optimise for change so it can be tweaked in the future if needed